x-pack/plugins/lens/server/saved_objects.ts

```diff
@@ -25,7 +29,10 @@ export function setupSavedObjects(core: CoreSetup) {
         uiCapabilitiesPath: 'visualize.show',
       }),
     },
-    migrations,
+    migrations: mergeMigrationFunctionMaps(
+      filterMigrations,
+      migrations as unknown as MigrateFunctionsObject
+    ),
```

Comment:
It's hard to know since there are currently no filter migrations, but this may not work. I would imagine that the filter migrations will expect to operate only on the filter objects. If that's the case, I assume we'd need to do something like this for each future lens migration function.
```js
doc.attributes.state.filters = doc.attributes.state.filters.map(filterMigrations['version-code'])
```

Reply:
Right, it won't work like this out of the box. Could you add this logic in a generic way? What I had in mind is to wrap the given `filterMigrations` in the way you suggest. We can test this on the unit level by passing in a sample filter migration that changes something (like negating it or something like this), then validating whether the resulting Lens saved object contains the expected filters.